Add gitignore.

mlucas opened the pull request:
This resolves https://github.com/Tino-FRC-2473/FRC2018-Offseason/issues/4.

Commit:
Add gitignore.

This resolves https://github.com/Tino-FRC-2473/FRC2018-Offseason/issues/4.

tnarayanan commented on .gitignore:
```diff
@@ -0,0 +1,148 @@
+*.log
+
+# BlueJ files
+*.ctxt
```
Not relevant because we are not using BlueJ

tnarayanan commented on .gitignore:
```diff
@@ -0,0 +1,148 @@
+*.ctxt
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
```
We are not using Mobile Tools for Java, so this is not necessary

tnarayanan reviewed (changes requested):
Some of the file extensions will not be used at all, so they do not need to be in the gitignore.

mvoodarla committed:
Remove all unnecessary ignores from gitignore

Removed gitignore for files like BlueJ ctxts as well as gitignores for
other languages such as Scala and PHP.

mvoodarla reviewed (approved):
I removed a few things that I thought were unnecessary so I think it's good to commit. We now need to remove all the files in the gitignore that have already been committed.
